src/main/java/robosky/uplands/entity/HexahaenEntity.java

```diff
@@ -0,0 +1,120 @@
+package robosky.uplands.entity;
+
+import net.minecraft.entity.*;
+import net.minecraft.entity.ai.goal.*;
+import net.minecraft.entity.attribute.EntityAttributes;
+import net.minecraft.entity.data.DataTracker;
+import net.minecraft.entity.data.TrackedData;
+import net.minecraft.entity.data.TrackedDataHandlerRegistry;
+import net.minecraft.entity.mob.HostileEntity;
+import net.minecraft.entity.player.PlayerEntity;
+import net.minecraft.nbt.CompoundTag;
+import net.minecraft.world.IWorld;
+import net.minecraft.world.LocalDifficulty;
+import net.minecraft.world.World;
+import scala.Int;
+
+import javax.annotation.Nullable;
+
+public class HexahaenEntity extends HostileEntity {
+
+    private static final TrackedData<Integer> STRENGTH;
+    public int strength;
+
+    static {
+        STRENGTH = DataTracker.registerData(HexahaenEntity.class, TrackedDataHandlerRegistry.INTEGER);
+    }
+
+    public HexahaenEntity(EntityType<? extends HexahaenEntity> entityType, World world) {
+        super(entityType, world);
+        this.strength = this.dataTracker.get(STRENGTH);
+    }
+
+    public HexahaenEntity(World world) {
+        this(EntityRegistry.HEXAHAEN, world);
+    }
+
+    @Override @Nullable
+    public EntityData initialize(IWorld world, LocalDifficulty difficulty, SpawnType spawnType, EntityData data, CompoundTag tag) {
+        setStrength(this.random.nextInt(5) + 1);
+        this.setHealth(this.getMaximumHealth());
+        super.initialize(world, difficulty, spawnType, data, tag);
+        return data;
+    }
+
+    @Override
+    protected void initGoals() {
+        this.goalSelector.add(4, new MeleeAttackGoal(this, 1.25, false));
+        this.goalSelector.add(5, new GoToWalkTargetGoal(this, 1.0));
+        this.goalSelector.add(7, new WanderAroundFarGoal(this, 1.0, 0.0f));
+        this.goalSelector.add(8, new LookAtEntityGoal(this, PlayerEntity.class, 8.0f));
+        this.goalSelector.add(8, new LookAroundGoal(this));
+        this.targetSelector.add(1, new RevengeGoal(this).setGroupRevenge());
+        this.targetSelector.add(2, new FollowTargetGoal<>(this, PlayerEntity.class, true));
+    }
+
+    @Override
+    protected void initDataTracker() {
+        super.initDataTracker();
+        this.dataTracker.startTracking(HexahaenEntity.STRENGTH, 1);
+    }
+
+    @Override
+    public void writeCustomDataToTag(CompoundTag tag) {
+        super.writeCustomDataToTag(tag);
+        tag.putInt("Strength", this.strength);
+    }
+
+    @Override
+    public void readCustomDataFromTag(CompoundTag tag) {
+        super.readCustomDataFromTag(tag);
+        tag.putInt("Strength", this.strength);
```

Comment:
Shouldn't this be `getInt`?

Reply:
I accidentally copied the `writeCustomDataToTag` method here. Fixed it